develop/hardware-interfaces/tutorial.md

```diff
@@ -4,4 +4,171 @@ title: Tutorial
 permalink: /docs/develop/hardware-interfaces/tutorial
 ---
 
-TODO
+## Creating a New Hardware Interface
+
+This tutorial will show you how to create a new Hardware Interface in an add-on.
+
+### Setting up a new add-on
+
+If you've already gone through the [Spatial Tool tutorial](../spatial-tools/tutorial), you can
+reuse the addon called `my-addon` that you created in that tutorial. If not, create a new
+directory called `my-addon` in the `vuforia-spatial-edge-server/addons` directory.
+
+Now, create a directory called `interfaces` in `my-addon`. Each directory within `interfaces`
+will be a separate hardware interface – a new "plugin" for this server to communicate with an
+external system.
+
+### Create the index.js file
+
+Create a directory in `interfaces` called `testInterface`, and then create a file called `index.js`
+inside `testInterface`. The index.js file of each hardware interface will get executed by the Edge
+Server when the Node.js environment finishes initializing the Edge Server's core capabilities.
+This file is isolated from the rest of the server with the exception of the server's
+`libraries/hardwareInterfaces.js` file, which provides an API that hardware interfaces can use
+to bridge the functionality of the server to external hardware.
+
+All hardware interfaces can be set up with the following template:
+
+```javascript
+// include the hardware interface API module from the server
+let server = require('@libraries/hardwareInterfaces');
+
+// load any persistent settings for this interface, such as whether it is enabled
+let settings = server.loadHardwareInterface(__dirname);
+
+// export these so that the hardware interface can be configured on the web interface
+exports.enabled = settings('enabled');
+exports.configurable = true;
+
+if (exports.enabled) {
+    // place your interface code here
+    // doing so properly ignores this interface if it has been disabled on the web interface
+}
+```
+
+### Using the Hardware Interface API
+
+Inside the template, you will need to access some combination of the Hardware Interface APIs
+based on what you want this interface to do.
+
+In general, implementing a hardware interface looks like this:
+
+1. Determine which nodes you want to connect to the hardware. This can be hard-coded or loaded
+from external settings.
+2. Implement a hardware-specific method for how to send data to that piece of hardware when data
+arrives at one of these nodes.
+3. Implement a hardware-specific method to listen to new values being generated by the hardware
+, and to send those values into the Spatial Toolbox system.
+
+The corresponding APIs, which are only a subset of the provided APIs but are the most common ones to
+use are:
+
+- [addNode](./api-reference#addNode)
```

Comment:
For these to work on the webpage the api-reference file will need a corresponding <a id="addNode"></a>

Reply:
Ah, right! I'll add those to the API reference so that this works, before I merge